### ISSUE TITLE
[Enhancement] Add additional configuration options to chickens stats (capping stats at a lower level and rate limiting cross bred chickens)

### ISSUE THREAD

**Issue by mahldcat** [setycz/ChickensMod#82]
Had a couple of configuration tweak ideas:  

**First** 

Have a Max stat limitation for each chicken...e.g. you can specify that diamond chickens cap out at say 2/2/2, or ender pearl chickens at 4/4/10.

**Second** 

Have some sort of cross breeding limiter (similar mechanic to agricraft cross breeding) for each stat (Growth/Gain/Strength), make this a decimal value between 0-1.

This limiter gets applied to the scores of the new egg produced by cross breeding:

Configuration value  (missing)  == keep the original calculation
Configuration value= 0 ==reset the given stat back to 1
Configuration value= 1 ==keep the originally calculated stat 
Configuration value between 0 & 1  ==  multiply the value to the calculated score and round it down

For instance if you had a limiter of say 
Growth = .5
Gain = 0
Strength = 1

When you breed two 10/10/10 chickens, instead you would end up with an egg that has the stats of 5/1/10.

**Comment by TokkenGmr** [comment]
I agree. An option in the config to handle breeding/mutation stats  would be amazing. It would allow pack creators more flexibility.

**Comment by ghost** [comment]
Perhaps and option to require minimum stats for cross-breading? Getting purple from red and blue chickens might only be 2/2/2 or 3/3/3, but breeding the diamond requires 10/10/10. Add the loss multiplier and you'll need to breed two 10/10/10 chickens to cross-breed a 1/1/1 diamond chicken. Now, that would be cruel, but still better than getting the base chickens to 10/10/10 and not sweating the rest.
